### ISSUE TITLE
Create a tool to extract a sub-sequence from a given fasta file.

### ISSUE THREAD

**Issue by jonn-smith** [broadinstitute/gatk#4202]
We should have a tool that takes in a FASTA file and an interval and creates another FASTA file that contains only the bases in that interval.  

This new FASTA file should be automatically indexed and a dictionary should be created for it.

**Comment by lbergelson** [comment]
The problem with this is that every fasta starts at position 1, so it would change the coordinates.  It would be great if we had a fasta specification that let you start at a position other than 1.

**Comment by jonn-smith** [comment]
True, but it only matters when the start position is not 1.  In that case I will append the range of the selected interval to the contig name so that it can't be confused with the whole contig.
